larsim/Utils/TruthMatchUtils.cc

```diff
@@ -0,0 +1,178 @@
+/**
+ * @file larsim/Utils/TruthMatchUtils.cc
+ *
+ * @brief Implementation of the TruthMatchUtils functions
+ *
+ * @author Dom Brailsford (d.brailsford@lancaster.ac.uk)
+ *
+ * $Log: $
+ */
+
+//STL
+#include <utility>
+//ART
+#include "canvas/Utilities/Exception.h"
+#include "messagefacility/MessageLogger/MessageLogger.h"
+//LARSoft
+#include "larsim/MCCheater/BackTrackerService.h"
+
+#include "TruthMatchUtils.h"
+
+namespace
+{
+
+using namespace TruthMatchUtils;
+IDToEDepositMap::const_iterator MaxEDepElementInMap(const IDToEDepositMap &idToEDepMap)
+{
+    IDToEDepositMap::const_iterator highestContribIt(std::max_element(idToEDepMap.begin(), idToEDepMap.end(),
+        [](const std::pair<G4ID, EDeposit>& a, const std::pair<G4ID, EDeposit>& b)->bool{ return
+        std::nextafter(a.second, std::numeric_limits<EDeposit>::lowest()) <= b.second && std::nextafter(a.second, std::numeric_limits<EDeposit>::max()) >= b.second
```

Comment:
What is the reason for using `std::nextafter`?

Reply:
It's for equality checking `a` and `b`.